### ISSUE TITLE
Containers that are updated via the docker-compose trigger lose their default service name alias

### ISSUE THREAD

**Issue by david-holbrook** [fmartinou/whats-up-docker#101]
I've noticed that when the docker-compose trigger updates a container, that container will get two new random container aliases set, rather than one random and one set to the service name from the compose file.

Example:

Compose file:

  radarr:
    image: linuxserver/radarr
    volumes:
      - /volume1/docker/radarr/config:/config
      - /volume1/docker/qbittorrent/downloads:/downloads
      - /volume1/movies:/movies
    environment:
      - PUID=xxxx
      - PGID=xxxx
      - TZ=Australia/Hobart
      - UMASK=022
    ports:
      - 7878:7878
    labels:
      - traefik.http.routers.radarr.entrypoints=https
      - traefik.http.routers.radarr.middlewares=organizr-auth@file
    restart: unless-stopped

Docker inspect result, when running docker compose manually.

                "Networks": {
                "containers": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "radarr",
                        "6573a8175ff6"**

Docker inspect result after WUD updates the container:

                "Networks": {
                "containers": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "f823ab916538",
                        "65b70c506495"**

This causes DNS to break within other containers on the same network, as containers are references by their service name alias. I could probably get around this by setting the container_name on each container, but this isn't very scalable and goes against the default behavior of docker.

**Comment by fmartinou** [comment]
Thanks for the precision of the issue context 👍!

I haven't seen this issue so far because I set names on my containers.

I'm going to see if I can clone alias properties as well as the other container properties.

**Comment by fmartinou** [comment]
Hi,

[Here is an attempt](https://github.com/fmartinou/whats-up-docker/commit/4525de7245ab723858f34731621a7c25e8b6dd49#diff-503942a2a397c9f848dff78acc8c292cc3d59562861b0d38a76abc0d65e889dbR193) to fix the issue.

Can you give it a try (`fmartinou/whats-up-docker:fix__101`) and tell me if everything is ok?

Thank you!

**Comment by david-holbrook** [comment]
Awesome - thank you!

I've just pulled the fix so once an update releases for any of my containers I will test and see 🤞

Will this clone all/both of the aliases? I could be mistaken but I think the "random" alias is generated from the container name which gets regenerated on each start. It looks to me like it's the first 12 characters of the container ID, so cloning this from a previous instance may cause other issues later on. If you do a docker inspect on any container and then search the output for the alias it appears 7 times throughout varias part of the container.

**Comment by fmartinou** [comment]
Hi,

Below the main difference between using Docker-Compose itself versus WUD.

## Initial situation
### docker-compose.yml
```yaml
version: '3'

services:
    traefik_test:
      image: traefik:2.4.5
```

### docker inspect
```
{
  ...
  "NetworkSettings": {
    ...
    "Networks": {
      "test_default": {
	    ...
        "Aliases": [
          "traefik_test",
          "a25bb4dc069a"
        ]
      }
    }
  }
}
```

## Next situation
### docker-compose.yml
```yaml
version: '3'

services:
    traefik_test:
      image: traefik:2.6.0
```

### alternative1: `docker inspect` after `docker-compose ` updated the service (`docker-compose up -d...`)
```
{
  ...
  "NetworkSettings": {
    ...
    "Networks": {
      "test_default": {
	    ...
        "Aliases": [
          "traefik_test",
          "e0aaf176122e"
        ]
      }
    }
  }
}
```

### alternative2: `docker inspect` after `wud` automatically updated the service
```
{
  ...
  "NetworkSettings": {
    ...
    "Networks": {
      "test_default": {
	    ...
        "Aliases": [
          "traefik_test",
          "e0aaf176122e",
          "03fddf3bb761"
        ]
      }
    }
  }
}
```

## To sum up:
The main difference is that the **`wud` clone operation keeps all previous aliases, including the random id one**.
In addition, a new "id alias" is also properly generated.

I don't think it will cause issues because everyone uses the "service name alias" and never the id one.
If I'm mistaken, it should be doable to remove the previous "id alias" when cloning the current container properties...

**Comment by fmartinou** [comment]
I just pushed a new version [where I remove the existing "service id alias"](https://github.com/fmartinou/whats-up-docker/commit/1cfbd99e47d4d34f9b04c1b5a9ce06bb6880182e#diff-503942a2a397c9f848dff78acc8c292cc3d59562861b0d38a76abc0d65e889dbR204).

It's a little bit cleaner ;)

**Comment by david-holbrook** [comment]
Nice one! I've just pulled it in, thanks.

It might be a silly idea; but if my compose file is only using the :latest tag on all services, is there any need for WUD to clone a new compose file at all? I've noticed it doesn't backup any version of my docker-compose.yml - since nothing gets changed when a container update is released when the :latest tag is always used. Could WUD just run docker-compose -up -d with my original compose file? Sorry - I might be going down a rabbit hole here... 😕

**Comment by fmartinou** [comment]
Unfortunately, wud doesn't use docker-compose at all (mainly because it's a Python CLI).

So it can just read the yaml, clone all containers with newer versions and write the yaml back .

**Comment by david-holbrook** [comment]
fix__101 has resolved the issue with my service aliases getting lost 😃 

I'm seeing the previous service id alias get preserved, but sounds like you have a fix for this ready to go.

"Aliases": 
   "prowlarr",
   "0f9216fc8cfd",
   "52fb5ebd5fcf"

Thanks a lot mate!

**Comment by fmartinou** [comment]
Great!

> I'm seeing the previous service id alias get preserved, but sounds like you have a fix for this ready to go.
Yes I fixed it.

I'm merging and closing the issue; the fix will be part of the next `wud` release.
Thanks for your interest!

**Comment by fmartinou** [comment]
The fix is available in version `5.15.0`
